lib/utils/__tests__/mailup.test.ts

```diff
@@ -13,9 +13,13 @@ afterEach(() => {
   jest.resetAllMocks();
 });
 
+// as superagent does not export request methods directly
+// we must override the superagent.Request prototype
+// so we can set up our jest mock to use it instead
+// of the send() method
 const mockSuperagentResponse = (response: any) =>
   // tslint:disable-next-line:no-object-mutation
-  (Object.getPrototypeOf(request("", "")).send = jest.fn()).mockReturnValueOnce(
+  ((superagent as any).Request.prototype.send = jest.fn()).mockReturnValueOnce(
```

Comment:
thx, I would just split this line for readability:

```ts
const sendMock = jest.fn();
(superagent as any).Request.prototype.send = sendMock;
sendMock.mockReturnValueOnce(...)
```

Reply:
another option is to just use
```ts
const getSendMock = () => (superagent as any).Request.prototype.send
```

then use it inside tests with 
```
getSendMock().mockReturnValue(...)
```

Reply:
`getSendMock` can be misleading as it creates a new mock each time, perhaps `newSendMock` or `initSendMock`

anyway, just make the code cleaner by splitting the creation of the mock from its usage

Reply:
done